### ISSUE TITLE
Module 'itertools' has no attribute 'imap'

### ISSUE THREAD

**Issue by crgwbr** [django-oscar/django-oscar-api#74]
The function `itertools.imap` was changed to just be `map` in Python 3 ([stackoverflow.com](https://stackoverflow.com/questions/30271712/i-cant-find-imap-in-itertools-in-python)). This breaks the basket list endpoint in Python 3.x.

[oscarapi/views/basic.py#L58](https://github.com/django-oscar/django-oscar-api/blob/master/oscarapi/views/basic.py#L58)

```
    def get_queryset(self):
        qs = super(BasketList, self).get_queryset()
        return itertools.imap(
            functools.partial(assign_basket_strategy, request=self.request),
            qs)
```

**Comment by maerteijn** [comment]
Will work on a fix

**Comment by maerteijn** [comment]
Fixed and released as version 1.0.6 https://pypi.python.org/pypi/django-oscar-api/1.0.6
